### PR TITLE
go.mod: update golang.org/x/net to v0.17.0

### PR DESCRIPTION
**Pull request by raghavendra-talur**
This is get the fixes for the three CVEs given below
CVE-2023-3978
CVE-2023-44487
CVE-2023-39325

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
Commands that I ran to get this change
```

File: go.mod [hunk 1]
```diff
@@ -134,14 +134,14 @@ require (
 	go.starlark.net v0.0.0-20221010140840-6bf6f0955179 // indirect
 	go.uber.org/atomic v1.10.0 // indirect
 	go.uber.org/multierr v1.9.0 // indirect
-	golang.org/x/crypto v0.6.0 // indirect
+	golang.org/x/crypto v0.14.0 // indirect
 	golang.org/x/mod v0.10.0 // indirect
-	golang.org/x/net v0.10.0 // indirect
+	golang.org/x/net v0.17.0 // indirect
```
Comment (nirs) [inline go.mod:139]: How indirect dependency changed if no direct dependency was changed?

**Review by nirs — COMMENTED** [review]
The second CVE looks like an error, how Go is related to swift?
https://github.com/advisories/GHSA-qppj-fm5r-hxr3

Is this based on bot pr from github?

File: go.mod (hunk 1 above)
Comment (ShyamsundarR) [reply go.mod:139]: @raghavendra-talur specifically ran go get for the required version that fixes the CVE as mentioned [here](https://github.com/RamenDR/ramen/pull/1100#issuecomment-1769270195) to update the indirect dep

**Review by ShyamsundarR — APPROVED** [review]

**Comment by raghavendra-talur** [comment]
> The second CVE looks like an error, how Go is related to swift? [GHSA-qppj-fm5r-hxr3](https://github.com/advisories/GHSA-qppj-fm5r-hxr3)
> 
> Is this based on bot pr from github?

Yes, this was from the github security notification.

**Pull request merged**